honeybee/model.py

```diff
@@ -0,0 +1,480 @@
+
+    @property
+    def rooms(self):
+        """A list of all Room objects in the model."""
+        return self._rooms.values()
+
+    @property
+    def faces(self):
+        """A list of all Face objects in the model."""
+        return self._faces.values()
+
+    @property
+    def shades(self):
+        """A list of all Shades objects in the model."""
+        return self._shades.values()
+
+    @property
+    def apertures(self):
+        """A list of all Aperture objects in the model."""
+        return self._apertures.values()
+
+    @property
+    def doors(self):
+        """A list of all Door objects in the model."""
+        return self._doors.values()
+
+    @property
+    def has_ngon_apertures(self):
+        """Boolean noting whether the model has apertures with more than 4 sides.
+
+        This is helpful for energy simulation since EnergyPlus cannot accept
+        sub-faces with more than 4 sides. This property should be checked before
+        running the triangulate_ngon_apertures() method since many models do not
+        have such apertures.
+        """
+        for ap in self.apertures:
+            if len(ap.geometry) > 4:
+                return True
+        return False
+
+    @property
+    def has_ngon_doors(self):
+        """Boolean noting whether the model has doors with more than 4 sides.
+
+        This is helpful for energy simulation since EnergyPlus cannot accept
+        sub-faces with more than 4 sides. This property should be checked before
+        running the triangulate_ngon_doors() method since many models do not have
+        such doors.
+        """
+        for dr in self.doors:
+            if len(dr.geometry) > 4:
+                return True
+        return False
+
+    def get_room(self, room_name):
+        """Get a room object in the model given the room name."""
+        return self._rooms[room_name]
+
+    def get_face(self, face_name):
+        """Get a face object in the model given the face name."""
+        return self._faces[face_name]
+
+    def get_shade(self, shade_name):
+        """Get a shade object in the model given the shade name."""
+        return self._shades[shade_name]
+
+    def get_aperture(self, aperture_name):
+        """Get an aperture object in the model given the aperture name."""
+        return self._apertures[aperture_name]
+
+    def get_door(self, door_name):
+        """Get an door object in the model given the door name."""
+        return self._doors[door_name]
+
+    def append_model(self, other_model):
+        """Append another Model object to this model."""
+        assert isinstance(other_model, Model), \
+            'Expected Model. Got {}.'.format(type(other_model))
+        for room in other_model.rooms:
+            self.append_room(room)
+        for face in other_model.faces:
+            if not face.has_parent:
+                self.append_face(face)
+        for shade in other_model.shades:
+            if not shade.has_parent:
```

Comment:
Shouldn't this always be the case as this shade is part of the model? Same comment about face and aperture.

Reply:
I see that you are collecting all shades, faces and apertures so now it makes sense but I'm not sure if we need to do that. If we have implemented access to faces effectively in room objects then this is the time to use them in my opinion. What I'm trying to say is that we can only collect orphaned faces under `_faces` and then for `faces` method we can collect all the faces from rooms and return them with orphaned faces. Something like this: `itertools.chain((room.faces for room in self.rooms), self._faces)`

Reply:
Let's have a longer call to discuss this after we have decided on the exact way that we want to organize the objects (if not using dictionaries).

Reply:
See my comment above about "Relying on parent objects to store the only reference to their children" for my thoughts on your proposal here.  If we think that we are planning on not using any of the checking methods, then what you propose here is worth considering.  But otherwise, the check for duplicate names and adjacent objects will have so much overhead.